### ISSUE TITLE
How to stop YouTube videos from reloading when any event occurs on the Slick carousel?

### ISSUE THREAD

**Issue by hmatt843** [kenwheeler/slick#1749]
I'm using the Slick carousel plugin to create a carousel of videos that users can add to, remove from, and filter. 
The problem I've got is that whenever a user does an event (add/remove/filter), all of the videos on the carousel reload, which gives a clunky feeling. And, if the user is playing a video, they obviously would not want their video to be interrupted whenever they did an action.

I found that the reloading can be stopped if I go into slick.js  (http://kenwheeler.github.io/slick/slick/slick.js) and comment out `_.$slidesCache = _.$slides;`, but that breaks the Filtering function.
Is there a way to prevent reloading while still preserving all functions?

JSFiddle: http://jsfiddle.net/neowot/eoov2ud1/37/ 

Javascript

    $(document).ready(function(){
        var slideId = 0;
        var slides = [];
        $('.add-remove').slick({
            slidesToShow: 3,
            slidesToScroll: 3
        });

        var target = '<div class="videowrapper"><iframe src="https://www.youtube.com/embed/7WgYmnwO-Pw" frameborder="0" allowfullscreen></iframe></div>';

        $('.js-add-FirstClass-slide').on('click', function() {        
            $('.add-remove').slick('slickAdd','<div class="FirstClass"  slide-id="' + slideId + '"><h3><a class="sliderX">X</a>' + target + '</h3></div>');
            slides.push(slideId);
            slideId++;
        });

        $('.js-add-SecondClass-slide').on('click', function() {
            $('.add-remove').slick('slickAdd','<div class="SecondClass" slide-id="' + slideId + '"><h3><a class="sliderX">X</a>' + target + '</h3></div>');
            slides.push(slideId);
            slideId++;
        });

        var filtered = false;

        $('.ToggleFirstClass').on('click', function() {
            if (filtered === false) {
                $('.add-remove').slick('slickFilter', $('.FirstClass')); 
                filtered = true;
            } else {
                $('.add-remove').slick('slickUnfilter');
                filtered = false;
            }
        });

        $('.ToggleSecondClass').on('click', function() {
            if (filtered === false) {
                $('.add-remove').slick('slickFilter','.SecondClass'); 
                filtered = true;
            } else {
                $('.add-remove').slick('slickUnfilter');
                filtered = false;
            }
        });

        $('body').on('click', '.sliderX', function() {
            var id = parseInt($(this).closest("div").attr("slide-id"), 0);
            var index = slides.indexOf(id);
            $('.add-remove').slick('slickRemove', index);
            slides.splice(index, 1);
        });
    });

slick.JS add event code

    Slick.prototype.addSlide = Slick.prototype.slickAdd = function(markup, index, addBefore) {
        var _ = this;

        if (typeof(index) === 'boolean') {
            addBefore = index;
            index = null;
        } else if (index < 0 || (index >= _.slideCount)) {
            return false;
        }

        _.unload();

        if (typeof(index) === 'number') {
            if (index === 0 && _.$slides.length === 0) {
                $(markup).appendTo(_.$slideTrack);
            } else if (addBefore) {
                $(markup).insertBefore(_.$slides.eq(index));
            } else {
                $(markup).insertAfter(_.$slides.eq(index));
            }
            } else {
            if (addBefore === true) {
                $(markup).prependTo(_.$slideTrack);
            } else {
                $(markup).appendTo(_.$slideTrack);
            }
        }

        _.$slides = _.$slideTrack.children(this.options.slide);

        _.$slideTrack.children(this.options.slide).detach();

        _.$slideTrack.append(_.$slides);

        _.$slides.each(function(index, element) {
            $(element).attr('data-slick-index', index);
        });

        _.$slidesCache = _.$slides;

        _.reinit();
    };

**Comment by leggomuhgreggo** [comment]
Interesting! I know this is from awhile back but it's a good issue. The trouble is the filter method uses the jQuery `.detach()` method, which removes the slides from the dom and then re-adds them. I think the `addSlide` and `removeSlide` methods might work better.

I'm gonna close this assuming it's no longer relevant, but holler back if you want to re-explore it.

**Comment by MobiusHorizons** [comment]
not sure if it is exactly the same issue, but slick seems to be re-adding the iframe of a youtube video   just by initializing slick slider. In the network tab, that iframe shows up three times, and two of them are initialized by jquery's append function. This seems like a bug.
